Mathlib/Logic/Basic.lean

```diff
@@ -1288,3 +1288,10 @@ theorem dite_prop_iff_and {Q : P → Prop} {R : ¬P → Prop} [Decidable P] :
   by_cases h : P <;> simp [h, forall_prop_of_false, forall_prop_of_true]
 
 end ite
+
+-- Aliases to work around lean4#2343
+alias Pi.inhabited := instInhabitedForAll_1
+#align pi.inhabited Pi.inhabited
+
+alias Pi.subsingleton := instSubsingletonForAll
+#align pi.subsingleton Pi.subsingleton
```

Comment:
What reducibility does `alias` generate? I think `export` might be more appropriate than `alias` here.

Reply:
I don't know how to use `export` to do this. Feel free to adjust.